engine/api/src/main/java/io/deephaven/engine/table/vectors/ColumnVectors.java

```diff
@@ -0,0 +1,489 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.vectors;
+
+import io.deephaven.engine.rowset.TrackingRowSet;
+import io.deephaven.engine.table.ColumnDefinition;
+import io.deephaven.engine.table.ColumnSource;
+import io.deephaven.engine.table.Table;
+import io.deephaven.vector.ByteVector;
+import io.deephaven.vector.CharVector;
+import io.deephaven.vector.DoubleVector;
+import io.deephaven.vector.FloatVector;
+import io.deephaven.vector.IntVector;
+import io.deephaven.vector.LongVector;
+import io.deephaven.vector.ObjectVector;
+import io.deephaven.vector.ShortVector;
+import io.deephaven.vector.Vector;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Utility methods for constructing {@link Vector Vectors} from the columns of a {@link Table}, enabling inefficient
+ * random-access and efficient bulk access to column data by row position.
+ */
+public final class ColumnVectors {
+
+    private ColumnVectors() {}
+
+    /**
+     * Get a {@link Vector} of the data belonging to the specified column. This is useful for randomly-accessing column
+     * data by row position (which is fairly inefficient), or extracting data from the column in bulk (which is quite
+     * efficient).
+     * <p>
+     * If {@code table} is {@link Table#isRefreshing() refreshing}, the returned vector is only valid for use during the
+     * current cycle.
+     * <p>
+     * Users should generally prefer one of the typed variants, e.g. {@link #ofChar ofChar} or {@link #ofObject
+     * ofObject}, rather than this method.
+     *
+     * @param table The {@link Table} to access column data from
+     * @param columnName The name of the column to access
+     * @return A {@link Vector} of the data belonging to the specified column
+     */
+    public static Vector<?> of(
+            @NotNull final Table table,
+            @NotNull final String columnName) {
+        return of(table, columnName, false);
+    }
```

Comment:
I wonder, do we _need_ to provide overloads without `usePreviousValues`? Does it make sense to cut the surface area in half?

Reply:
IMO, yes. I want users to be able to pretend that option doesn't exist.